### PR TITLE
Ports most of geospace module to 010

### PR DESCRIPTION
**Pull request by vertexmc**

### CODE REVIEW

File: synapse/models/geospace.py [hunk 1]
```diff
@@ -1,85 +1,109 @@
...
+
+class LatLong(s_types.Type):
+    # FIXME nothing uses these props
+    # FIXME should these be indexed as something other than a string?
+
+    def postTypeInit(self):
+        self.setNormFunc(str, self._normPyStr)
 
-    def norm(self, valu, oldval=None):
+    def _normPyStr(self, valu):
```
Comment (vertexmc) [inline synapse/models/geospace.py]: Probably also need a norm for tuple?

File: synapse/models/geospace.py [hunk 2]
```diff
@@ -1,85 +1,109 @@
...
+            raise s_exc.BadTypeValu(valu, mesg='Latitude may only be -90.0 to 90.0')
+        if lonv > 180 or lonv < -180:
+            raise s_exc.BadTypeValu(valu, mesg='Longitude may only be -180.0 to 180.0')
 
-        except Exception as e:
-            self._raiseBadValu(valu, mesg='Invalid float format')
+        return (latv, lonv), {'subs': {'lat': latv, 'lon': lonv}}
 
-        #TODO eventually support minutes / sec and N/S E/W syntax
+    def indx(self, valu):
```
Comment (vertexmc) [inline synapse/models/geospace.py]: Is this how we want to index latlong pairs?

File: synapse/models/geospace.py [hunk 3]
```diff
@@ -1,85 +1,109 @@
...
+        # to unpack:
```
Comment (vertexmc) [inline synapse/models/geospace.py]: Would this unpack be used in special lifts?

File: synapse/models/geospace.py [hunk 4]
```diff
@@ -1,85 +1,109 @@
...
+        # latv, lonv = struct.unpack('>qq', norm)
+        # latv, lonv = latv / 10**8, lonv / 10**8
+        return struct.pack('>qq', int(valu[0] * 10**8), int(valu[1] * 10**8))
 
-        if latv > 90 or latv < -90:
-            self._raiseBadValu(valu, mesg='Latitude may only be -90.0 to 90.0')
+    def repr(self, norm):
```
Comment (vertexmc) [inline synapse/models/geospace.py]: I think this is what we want the repr to be

File: synapse/models/geospace.py [hunk 5]
```diff
@@ -1,85 +1,109 @@
...
-            latv = float(lat.strip())
-            lonv = float(lon.strip())
+        if latv > 90 or latv < -90:
+            raise s_exc.BadTypeValu(valu, mesg='Latitude may only be -90.0 to 90.0')
+        if lonv > 180 or lonv < -180:
+            raise s_exc.BadTypeValu(valu, mesg='Longitude may only be -180.0 to 180.0')
 
-        except Exception as e:
-            self._raiseBadValu(valu, mesg='Invalid float format')
+        return (latv, lonv), {'subs': {'lat': latv, 'lon': lonv}}
```
Comment (vertexmc) [inline synapse/models/geospace.py]: Should the norm really be a tuple of floats? Should the subs be str reprs?

File: synapse/models/geospace.py (hunk 1 above)
Comment (invisig0th) [reply synapse/models/geospace.py]: Definitely. In cases like this, tuple and list should both be handled.

File: synapse/models/geospace.py (hunk 5 above)
Comment (invisig0th) [reply synapse/models/geospace.py]: Maybe we need a type for the individual fields ( so they can be scaled correctly in single use? )  perhaps a math "degrees" type that always express degrees in trig terms?  Hrm.  Lemme ponder the subs.  The main value should def be (<float>,<float>) ( obvs with special indx() value ) but lemme think about subs. :D

File: synapse/models/geospace.py [hunk 6]
```diff
@@ -1,85 +1,109 @@
...
 
-        #TODO eventually support minutes / sec and N/S E/W syntax
+    def indx(self, valu):
+        # to unpack:
+        # latv, lonv = struct.unpack('>qq', norm)
+        # latv, lonv = latv / 10**8, lonv / 10**8
+        return struct.pack('>qq', int(valu[0] * 10**8), int(valu[1] * 10**8))
```
Comment (invisig0th) [inline synapse/models/geospace.py]: There's still a trick here about signdness.  btree range indexing over signed values requires a scale up.  lets chat in a go-back tomorrow :)

File: synapse/models/geospace.py (hunk 3 above)
Comment (invisig0th) [reply synapse/models/geospace.py]: Maybe?  Probably not...

File: synapse/models/geospace.py (hunk 4 above)
Comment (invisig0th) [reply synapse/models/geospace.py]: Completely agreed.

File: synapse/models/inet.py [hunk 7]
```diff
@@ -572,10 +572,9 @@ def getModelDefs(self):
                             'doc': 'The ASN to which the IPv4 address is currently assigned.'
                         }),
 
-                        # FIXME implement geospace...
-                        #('latlong', ('geo:latlong', {}), {
-                        #    'doc': 'The last known latitude/longitude for the node'
-                        #}),
+                        ('latlong', ('geo:latlong', {}), {
+                            'doc': 'The last known latitude/longitude for the node'
```
Comment (invisig0th) [inline synapse/models/inet.py]: "The best known latitude,longitude for the IPv4 node."

**Comment by vertexmc** [comment]
This is ready for review again

**Pull request merged**